packages/discovery-react-components/src/components/DocumentPreview/components/PdfHighlight/PdfHighlight.tsx

```diff
@@ -164,21 +189,47 @@ const Highlight: FC<{
     });
   };
 
+  // This enrichment is encompassed by an overlap
+  const hasOverlap = overlapMeta.fieldIdWithOverlap.has(shape.highlightId);
+
+  // Style without facets
+  let customStyles = [
+    `${base}__item`,
+    className,
+    shape.className,
+    active && `${base}__item--active`,
+    active && activeClassName
+  ];
+  // Style without facets
+  if (shape.facetId) {
+    if (hasOverlap) {
+      // Enrichment is part of overlap, display only when active,
+      // otherwise overlap has precedent to be displayed
+      customStyles = customStyles.concat([
```

Comment:
Nit: a bit cleaner to just use `push()`:

```js
customStyles.push(
  baseOverlapHighlight,
  active && baseHighlightColorActive,
  ...
);
```

Here and below.

Reply:
done